src/librustc/traits/project.rs

```diff
@@ -1247,6 +1247,34 @@ fn confirm_callable_candidate<'cx, 'gcx, 'tcx>(
                                               fn_sig,
                                               flag);
 
+    if selcx.projection_mode().is_any() {
```

Comment:
I actually told @pnkfelix about `ProjectionMode::Any`, he had some finer-grained flag initially.

I'm not sure I understand what situations this hack can produce invalid results in.
Remember that trans doesn't just look for things that are *guaranteed* to succeed, but also has to be able to correctly tell if something *would succeed* (for various reasons pointed out in a comment above).